src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts

```diff
@@ -92,6 +93,23 @@ export default class NScaleVideoUplinkBandwidthPolicy implements VideoUplinkBand
     return Math.trunc(rate);
   }
 
+  scaleResolutionDownBy(): number {
+    if (this.hasBandwidthPriority) {
+      return 1;
+    }
+    let scale;
+    if (this.numParticipants <= 4) {
+      scale = 1;
+    } else if (this.numParticipants <= 8) {
+      scale = 1.5;
+    } else if (this.numParticipants <= 16) {
+      scale = 2;
+    } else {
+      scale = 4;
+    }
+    return scale;
+  }
```

Comment:
Does it make sense to make this an abstract policy, so you could write

```typescript
class DefaultNScaleVideoUplinkBandwidthPolicy extends AbstractNScaleVideoUplinkBandwidthPolicy {
  scaleResolutionDownBy(): number {
    return …
  }
}
```

or to have the policy take a pure function of `numParticipants` as a constructor argument?

Reply:
If I understand correctly, you want to expose the scale down and bitrate calculation logic so builders can pass their own version? If so, our current recommendation is to copy the downlink policy to use their own logic.